Mypy complains about numpy typings in src/ewatercycle/util.py

When I run `mypy src/ewatercycle/util.py ` I get

```
src/ewatercycle/util.py:35: error: Argument 1 to "meshgrid" has incompatible type "Iterable[float]"; expected "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]"
src/ewatercycle/util.py:35: error: Argument 2 to "meshgrid" has incompatible type "Iterable[float]"; expected "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]"
src/ewatercycle/util.py:48: error: No overload variant of "diff" matches argument type "Iterable[float]"
src/ewatercycle/util.py:48: note: Possible overload variants:
src/ewatercycle/util.py:48: note:     def [_T] diff(a: _T, n: Literal[0], axis: SupportsIndex = ..., prepend: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ..., append: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ...) -> _T
src/ewatercycle/util.py:48: note:     def diff(a: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], n: int = ..., axis: SupportsIndex = ..., prepend: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ..., append: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ...) -> ndarray[Any, dtype[Any]]
src/ewatercycle/util.py:49: error: No overload variant of "diff" matches argument type "Iterable[float]"
src/ewatercycle/util.py:49: note: Possible overload variants:
src/ewatercycle/util.py:49: note:     def [_T] diff(a: _T, n: Literal[0], axis: SupportsIndex = ..., prepend: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ..., append: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ...) -> _T
src/ewatercycle/util.py:49: note:     def diff(a: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]], n: int = ..., axis: SupportsIndex = ..., prepend: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ..., append: Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]] = ...) -> ndarray[Any, dtype[Any]]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ Formatted as described on [https://keepachangelog.com](https://keepachangelog.co
 ### Fixed
 
 - Test suite fails with fresh conda env ([#275](https://github.com/eWaterCycle/ewatercycle/issues/275))
+- incompatible numpy typings ([#285](https://github.com/eWaterCycle/ewatercycle/issues/285))
 
 ## [1.1.3] (2021-10-18)
 

--- a/src/ewatercycle/util.py
+++ b/src/ewatercycle/util.py
@@ -31,8 +31,11 @@ def find_closest_point(
         idx_lon: index of closest grid point in the original longitude array
         idx_lat: index of closest grid point in the original latitude array
     """
+    grid_longitudes_array = np.array(grid_longitudes)
+    grid_latitudes_array = np.array(grid_latitudes)
+
     # Create a grid from coordinates (shape will be (nlat, nlon))
-    lon_vectors, lat_vectors = np.meshgrid(grid_longitudes, grid_latitudes)
+    lon_vectors, lat_vectors = np.meshgrid(grid_longitudes_array, grid_latitudes_array)
 
     dlon = np.radians(lon_vectors - point_longitude)
     dlat = np.radians(lat_vectors - point_latitude)
@@ -45,8 +48,8 @@ def find_closest_point(
     distance = distances.min()
 
     # Rough check to see if point is in or near the grid
-    dx = np.diff(grid_longitudes).max() * 111  # (1 degree ~ 111km)
-    dy = np.diff(grid_latitudes).max() * 111  # (1 degree ~ 111km)
+    dx = np.diff(grid_longitudes_array).max() * 111  # (1 degree ~ 111km)
+    dy = np.diff(grid_latitudes_array).max() * 111  # (1 degree ~ 111km)
     if distance > max(dx, dy) * 2:
         raise ValueError(f"Point {point_longitude, point_latitude} outside model grid.")
 

